go/pkg/cas/upload.go

```diff
@@ -395,7 +395,13 @@ func (c *Client) Upload(ctx context.Context, opt UploadOptions, inputC <-chan *U
 		}
 	})
 
-	return &UploadResult{Stats: u.stats, u: u}, errors.WithStack(eg.Wait())
+	err := eg.Wait()
+	if err != nil {
+		// Caller may not unwrap stack frame of github.com/pkg/error, so log error here.
+		log.Errorf("upload err %+v", err)
+	}
+
+	return &UploadResult{Stats: u.stats, u: u}, err
```

Comment:
Why not have the callsite log the error? Modifying a common library seems more intrusive than the less global callsite.

Reply:
Are you suggesting to support stack extraction from github.com/pkg/errors in go.chromium.org/luci/common/errors?

Reply:
no? I mean, this code returns the exactly same error object that it logs. It means the callsite can do the same.

Reply:
I see the latest version now returns an error with a stack annotation.
I am not sure what's the problem logging with stack, but perhaps it does not matter much.

Reply:
I understood what you mean. Thanks.